### ISSUE TITLE
Deleting applications in cluster mode do not cleans data in MongoDB

### ISSUE THREAD

**Issue by medzin** [ant-media/Ant-Media-Server#3531]
### Short description

Deleting applications in cluster mode do not cleans data and collections in MongoDB which causes problems as managed MongoDB instances (e.g. Atlas) have limits for databases and collections. This is especially annoying on test environments when we continuously delete and create applications.

### Environment

* Operating system and version: Ubuntu 20.04
* Java version: openjdk 11.0.11 2021-04-20
* Ant Media Server version: Enterprise Edition 2.4.0-SNAPSHOT 20210810_0917
* Browser name and version: Microsoft Edge Version 92.0.902.67 (Official build) (64-bit)

### Steps to reproduce
1. Create an app in cluster mode via the script
2. Delete the same app in cluster mode via the script

### Expected behavior
* Document with application settings is removed from `clusterdb.AppSettings` collection
* `appname*` databases are removed from MongoDB server (maybe there should be a flag to specify that I want to remove the data from DB as well?)

### Actual behavior
All data stays in database.

**Comment by mekya** [comment]
Hi @medzin ,

It's an expected behavior. We just set the flag `toBeDeleted` to `true` when an application is deleted. We just synch(delete) the whole nodes in the cluster by using this field. 

Regards,
A. Oguz

**Comment by medzin** [comment]
Can I create an application with the same name again? When the databases and collections created by the application itself will be deleted (as there is quite a lot of them)?

**Comment by mekya** [comment]
Hi @medzin ,

Yes you can create the application again with the same name. There was a problem in creating the application with the same name on the same instance. It's fixed. 

I think if we delete the collections(except application settings) on the mongodb when we delete the application, you're happy with that?

**Comment by medzin** [comment]
@mekya yes - deleting the app collections will help a lot as this part was causing problems on Atlas MongoDB.

**Comment by mekya** [comment]
Ok. Added to the backlog.

**Comment by mekya** [comment]
Closing the issue because it's merged.
